packages/components/src/Menu/Menu.tsx

```diff
@@ -63,38 +63,53 @@ export interface SectionProps {
 // eslint-disable-next-line max-statements
 export function Menu({ activator, items }: MenuProps) {
   const [visible, setVisible] = useState(false);
-  const fullWidth = activator?.props?.fullWidth || false;
-  const [position, setPosition] = useState<Position>({
-    vertical: "below",
-    horizontal: "right",
-  });
-  const wrapperRef = createRef<HTMLDivElement>();
+  const shadowRef = useRef<HTMLSpanElement>(null);
+  const menuRef = useFocusTrap<HTMLDivElement>(visible);
+
+  const { width } = useWindowDimensions();
+
   const buttonID = useId();
   const menuID = useId();
 
+  const fullWidth = activator?.props?.fullWidth || false;
+
+  const wrapperClasses = classnames(styles.wrapper, {
+    [styles.fullWidth]: fullWidth,
+  });
+
   useOnKeyDown(handleKeyboardShortcut, ["Escape"]);
-  useSafeLayoutEffect(() => {
-    if (wrapperRef.current) {
-      const bounds = wrapperRef.current.getBoundingClientRect();
-      const newPosition = { ...position };
-
-      if (bounds.top <= window.innerHeight / 2) {
-        newPosition.vertical = "below";
-      } else {
-        newPosition.vertical = "above";
-      }
-
-      if (bounds.left <= window.innerWidth / 2) {
-        newPosition.horizontal = "right";
-      } else {
-        newPosition.horizontal = "left";
-      }
-
-      setPosition(newPosition);
-    }
-  }, [visible, fullWidth]);
   useRefocusOnActivator(visible);
 
+  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
+  const { styles: popperStyles, attributes } = usePopper(
+    shadowRef.current?.nextElementSibling,
+    popperElement,
+    {
+      placement: "bottom-start",
+      modifiers: [
+        {
+          name: "flip",
+          options: {
+            flipVariations: true,
+          },
+        },
+        {
+          name: "offset",
+          options: {
+            offset: [0, 2],
```

Comment:
2 is a bit magic, but it matches the old layout (right) exactly
<img width="498" alt="image" src="https://github.com/GetJobber/atlantis/assets/11843215/6e48e4b6-1e81-4536-b457-b2be92a9f128">

noticing the width of these is different though. have to look into that.

Reply:
2 questions

- can we make the offset 6? it's still a bit magic, but our [Figma mocks](https://www.figma.com/file/rIIhulZvcp9M82lNOCGv16/Product%2FOnline?type=design&node-id=14110%3A0&mode=design&t=dN7Zpe2AQMGHYuxn-1) have it set as 6

Reply:
yup. most definitely. this is how that would look

![image](https://github.com/GetJobber/atlantis/assets/11843215/49f7fc27-cdbd-4d59-90a1-1977137e949f)


haha that's actually what that `motion.div` code I was asking you about was trying to do. it's been broken for a few years now, but once upon a time that's what was happening. I have an idea of how to fix it. I'll give that a shot.

Reply:
should be working now where it'll feel like it is coming out of the button and back in based on the direction

Reply:
Nice! I'll give it a try when the build is up